x-pack/plugins/security_solution/public/resolver/view/resolver_without_providers.tsx

```diff
@@ -0,0 +1,136 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+/* eslint-disable no-duplicate-imports */
+
+/* eslint-disable react/display-name */
+
+import React, { useContext, useCallback } from 'react';
+import { useSelector } from 'react-redux';
+import { useEffectOnce } from 'react-use';
+import { EuiLoadingSpinner } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n/react';
+import * as selectors from '../store/selectors';
+import { EdgeLine } from './edge_line';
+import { GraphControls } from './graph_controls';
+import { ProcessEventDot } from './process_event_dot';
+import { useCamera } from './use_camera';
+import { SymbolDefinitions, useResolverTheme } from './assets';
+import { useStateSyncingActions } from './use_state_syncing_actions';
+import { useResolverQueryParams } from './use_resolver_query_params';
+import { StyledMapContainer, StyledPanel, GraphContainer } from './styles';
+import { entityId } from '../../../common/endpoint/models/event';
+import { SideEffectContext } from './side_effect_context';
+import { ResolverProps } from '../types';
+
+/**
+ * The highest level connected Resolver component. Needs a `Provider` in its ancestry to work.
+ */
+export const ResolverWithoutProviders = React.memo(
+  React.forwardRef(function (
+    { className, databaseDocumentID, resolverComponentInstanceID }: ResolverProps,
+    refToForward
+  ) {
+    /**
+     * This is responsible for dispatching actions that include any external data.
+     * `databaseDocumentID`
+     */
+    useStateSyncingActions({ databaseDocumentID, resolverComponentInstanceID });
+
+    const { timestamp } = useContext(SideEffectContext);
+
+    // use this for the entire render in order to keep things in sync
+    const timeAtRender = timestamp();
+
+    const { processNodePositions, connectingEdgeLineSegments } = useSelector(
+      selectors.visibleNodesAndEdgeLines
+    )(timeAtRender);
+    const terminatedProcesses = useSelector(selectors.terminatedProcesses);
+    const { projectionMatrix, ref: cameraRef, onMouseDown } = useCamera();
+
+    const ref = useCallback(
```

Comment:
ref needs more complex handling now. When we get a new dom ref, call `cameraRef`, which sets the useCamera ref. Then if we're dealing w/ a forward ref, handle that.